.github/workflows/CI.yml

```diff
@@ -557,8 +564,15 @@ jobs:
       ec2-matrix-2: ${{ steps.set-matrix.outputs.ec2-matrix-2 }}
       ec2-matrix-3: ${{ steps.set-matrix.outputs.ec2-matrix-3 }}
     steps:
-      - name: Checkout
+      - name: Checkout main
+        if: ${{ github.event.inputs.sha == null }}
+        uses: actions/checkout@v2
+
+      - name: Checkout sha
+        if: ${{ github.event.inputs.sha != null }}
```

Comment:
Do you need to do this in every checkout step?

Reply:
Ex here https://github.com/aws-observability/aws-otel-collector/pull/794/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8L599

Reply:
At first I thought the following EC2 test are not needed to checkout for the following ref. But after considering, I would use the set-output https://github.com/aws-observability/aws-otel-collector/pull/794/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8R482-R492 to avoid repeating these unnecessary steps.